shared/js/passcode_helper.js

```diff
@@ -10,154 +10,128 @@
+      return _deriveBits(pwKey, salt, iterations, algorithm);
+    });
+    promise = promise.catch((error) => {
+      console.error('PasscodeHelper: _derive_bits() failed!', error);
+      throw error;
+    });
+    return promise;
+  }
 
-  PasscodeHelper.prototype.setPasscode = function(newPass) {
-      var lock = navigator.mozSettings.createLock();
-      var getFromSettings = Promise.all([
-        lock.get(SET_DIGEST_ITERATIONS),
-        lock.get(SET_DIGEST_ALGORITHM)
-      ]);
-      var digest = getFromSettings.then((values) => {
-        // Always generate a new salt.
-        this._salt = crypto.getRandomValues(new Uint8Array(8));
-        /* the combined lock.get() makes it quite ugly. we get
-         an Array of objects, each with just one key/value,
-         which is the requested setting. let's destruct as follows:
-         */
-        var [digestIteration, digestAlgorithm] = values;
-        this._iterations = parseInt(digestIteration[SET_DIGEST_ITERATIONS]) ||
-          DEFAULT_ITERATIONS;
-        if (typeof(digestAlgorithm[SET_DIGEST_ALGORITHM]) === 'string') {
-          this._algorithm = digestAlgorithm[SET_DIGEST_ALGORITHM];
-        }
-        else {
-          this._algorithm = DEFAULT_ALGORITHM;
-        }
-        return this._makeDigest(newPass, this._salt,
-                                this._iterations, this._algorithm)
-        .then((digest) => { this._storeNewDigest(digest); }).catch((error) => {
-            console.error('PasscodeHelper: Could not make digest:', error);
-            throw error;
-          });
-      }).catch((error) => {
-        console.error('PasscodeHelper: No Settings?', error);
-        throw error;
-      });
-      return digest;
-    };
-  PasscodeHelper.prototype._storeNewDigest = function(digest) {
-      // Note: We can now store the salt, since digest was generated!
-      var digestUint8 = new Uint8Array(digest);
-      var newSettings = {};
-      newSettings[SET_DIGEST_SALT] = this._salt;
-      newSettings[SET_DIGEST_VALUE] = digestUint8;
-      newSettings[SET_DIGEST_ITERATIONS] = this._iterations;
-      newSettings[SET_DIGEST_ALGORITHM] = this._algorithm;
-      var lock = navigator.mozSettings.createLock();
+  function setPasscode(newPass) {
+    // Always generate a new salt.
+    var salt = crypto.getRandomValues(new Uint8Array(8));
+    /* the combined lock.get() makes it quite ugly. we get
+     an Array of objects, each with just one key/value,
+     which is the requested setting. let's destruct as follows:
+     */
+    var promise = _makeDigest(newPass, salt, DEFAULT_ITERATIONS,
+                              DEFAULT_ALGORITHM);
+    promise = promise.then((digest) => {
+      return _storeNewDigest(digest, salt);
+    });
+    promise = promise.catch((error) => {
+      console.error('PasscodeHelper: Could not make digest:', error);
+      throw error;
+    });
+    return promise;
+  }
+  function _storeNewDigest(digest, salt) {
```

Comment:
Would "storePasscode()" be a better name? After all we're storing the digest and salt.